services/npm/npm-rating.service.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
```

Comment:
I'm not crazy about the file naming and framing this as npm since the service is npms.io, not npmjs.com. It's important in the badge routes, code structure, etc. that we're very explicit about the corresponding service, and unlike our existing npm badges (the ones under the `services/npm` directory), this one is not from the npm registry.

Reply:
In fact, npmjs.com is using the score service and search service which provide by npms.io

and I'm worried about the typo of `npms` vs `npm` by user.

so that's why I put it to `services/npm`,

anyway, I'm open to moving to another directory, what's your options?

![image](https://user-images.githubusercontent.com/227713/121985390-3e291d80-cdc7-11eb-982d-ce85a0247db4.png)

Reply:
> In fact, npmjs.com is using the score service and search service which provide by npms.io

A few things to keep in mind:

* npms.io is the provider/upstream service of the data from Shields' perspective
* The npm registry is _currently_ a consumer of some, but not all, of that data
* We don't really concern ourselves with who other consumers of a service are, nor what other systems that service integrates with behind the scenes (even if it's well known information)
* npm could always change their provider (or user their own internal implementation), or drop it altogether.
* We have users that leverage self-hosted Shields to be able to generate badges for private packages on the npm registry, and these new badges utilizing npms won't work for private packages, but it would be a potential source of confusion if we made it seem like these new badges were native to the npm registry
    * In a similar vein, folks can self-host their own registry instance too which I also believe npms cannot support 

Users need to go the npms to see the raw data, including things like `final` which AFAIK is not something the npm registry has even incorporated.

> and I'm worried about the typo of npms vs npm by user.

Could you elaborate on your concern?

Badge urls tend to be a one time action for users, not something they type on a regular bases. The only time one would be type this is when using the Shields.io site to search for badges, in which cases typing `npm` would list all the badges (including the new ones proposed in this PR), and a user typing `npms` would just see the new badges. Additionally, the keywords defined on the examples in the service class can be used to add additional search terms, which would eliminate any typo risk in my mind.

Shields.io also supports building and exporting badge urls, but even in the event of a user handcrafting a url and having a typo, there is a loud indicator and an easy fix.

Reply:
ok, I will move it to `npms-io`